plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -62,11 +62,10 @@ export class ArgoService implements ArgoServiceApi {
             token,
           );
         } catch (error: any) {
-          this.logger.error(
-            `failed to fetch app data from ${argoInstance.name}: ${String(
-              error,
-            )}`,
-          );
+          return null;
+        }
+
+        if (options.selector && !getArgoAppDataResp.items) {
```

Comment:
I'm not familiar with the API. The second part check if the items key exists on the getArgoAppDataResp ? Is this the intended check or you'd like to check if the items array is actually empty? or both ?

Reply:
Yes. That is intended. When using selectors, the items-property of the returned JSON-object will be undefined, not an empty array. So in that case, trying to iterate over the elements of items would yield nothing but an error.